src/cargo/sources/path.rs

```diff
@@ -255,10 +255,13 @@ impl<'cfg> PathSource<'cfg> {
         }
         for dir in try!(fs::read_dir(path)) {
             let dir = try!(dir).path();
-            match (is_root, dir.file_name().and_then(|s| s.to_str())) {
-                (_,    Some(".git")) |
-                (true, Some("target")) |
-                (true, Some("Cargo.lock")) => continue,
+            let name = dir.file_name().and_then(|s| s.to_str());
+            let is_hidden = name.and_then(|s| Some(s.starts_with(".")));
+            // Skip cargo artifacts and hidden/dot directories
+            match (is_root, name, is_hidden) {
+                (_, _, Some(true)) |
+                (true, Some("target"), Some(false)) |
+                (true, Some("Cargo.lock"), Some(false)) => continue,
```

Comment:
This may actually be getting to the point where it'll look better broken up:

```rust
let name = dir.file_name().and_then(|s| s.to_str());

// skip dotfile directories
if name.map(|s| s.starts_with(".")) == Some(true) {
    continue
} else if is_root {
    // skip cargo artifacts
    match name {
        "target" | "Cargo.lock" => continue,
        _ => {}
    }
}
```